org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/generators/xml2xsd/XML2XMLSchemaGeneratorTest.java

```diff
@@ -74,7 +74,60 @@ public void schemaWithNS() throws IOException {
 	}
 
 	@Test
-	public void schemaWithAttr() throws IOException {
+	public void schemaWithDecimalAttr() throws IOException {
+		String xml = "<note version=\"1.2\" />";
+		String xsd = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + lineSeparator() + //
+				"<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">" + lineSeparator() + //
+				"  <xs:element name=\"note\">" + lineSeparator() + //
+				"    <xs:complexType>" + lineSeparator() + //
+				"      <xs:attribute name=\"version\" type=\"xs:decimal\" use=\"required\" />" + lineSeparator() + //
+				"    </xs:complexType>" + lineSeparator() + //
+				"  </xs:element>" + lineSeparator() + //
+				"</xs:schema>";
+		assertGrammarGenerator(xml, new XMLSchemaGeneratorSettings(), xsd);
+	}
+
+	@Test
+	public void schemaWithDecimalAttrAndContent() throws IOException {
+		String xml = "<note version=\"1.2\" >ABCD</note>";
+		String xsd = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + lineSeparator() + //
+				"<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">" + lineSeparator() + //
+				"  <xs:element name=\"note\">" + lineSeparator() + //
+				"    <xs:complexType>" + lineSeparator() + //
+				"      <xs:simpleContent>" + lineSeparator() + //
+				"        <xs:extension base=\"xs:string\">" + lineSeparator() + //
+				"          <xs:attribute name=\"version\" type=\"xs:decimal\" use=\"required\" />" + lineSeparator() + //
+				"        </xs:extension>" + lineSeparator() + //
+				"      </xs:simpleContent>" + lineSeparator() + //
+				"    </xs:complexType>" + lineSeparator() + //
+				"  </xs:element>" + lineSeparator() + //
+				"</xs:schema>";
+		assertGrammarGenerator(xml, new XMLSchemaGeneratorSettings(), xsd);
+	}
+
+	@Test
+	public void schemaWithDecimalAttrAndMixedContent() throws IOException {
+		String xml = "<note version=\"1.2\" >AB<C/>D</note>";
+		String xsd = "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" + lineSeparator() + //
+				"<xs:schema xmlns:xs=\"http://www.w3.org/2001/XMLSchema\">" + lineSeparator() + //
+				"  <xs:element name=\"note\">" + lineSeparator() + //
+				"    <xs:complexType mixed=\"true\">" + lineSeparator() + //
+				"      <xs:sequence>" + lineSeparator() + //
+				"        <xs:element name=\"C\" />" + lineSeparator() + //
+				"      </xs:sequence>" + lineSeparator() + //
+				"      <xs:simpleContent>" + lineSeparator() + //
+				"        <xs:extension base=\"xs:string\">" + lineSeparator() + //
+				"          <xs:attribute name=\"version\" type=\"xs:decimal\" use=\"required\" />" + lineSeparator() + //
+				"        </xs:extension>" + lineSeparator() + //
+				"      </xs:simpleContent>" + lineSeparator() + //
+				"    </xs:complexType>" + lineSeparator() + //
+				"  </xs:element>" + lineSeparator() + //
+				"</xs:schema>";
+		assertGrammarGenerator(xml, new XMLSchemaGeneratorSettings(), xsd);

```

Comment:
This XSD contains an error:
![image](https://user-images.githubusercontent.com/20326645/85609373-ff351180-b623-11ea-913b-0dac7a1e1440.png)

Reply:
fixed